mantidproject/mantid
# Indirect - QuickRun Algorithm

MSG have asked for an EnergyWindowScan algorithm that reduces .raw files and then runs them through ElasticWindowMultiple and then an option for MSDFit. The aim of this is to speed up the workflow of Indirect algorithms. Also in this algorithm the Elastic Incoherent Structure Factor is calculated from the elastic and inelastic ranges.
EnergyWindowScan is a general algorithm that requires all the input parameters for EnergyTransfer, IndirectEnergyWindowScan is a bit more customised, this is the basis for implementation of an interface.

**To test:**

Algorithms > EnergyWindowScan 
InputFiles `irs26176`, Instrument=`IRIS`, Analyser=`graphite`, Reflection=`002`, SpectraRange=`3, 50`, ElasticRange=`-0.5, 0`, InelasticRange=`0, 0.5`, GroupingMethod=`All`

Repeat for IndirectEnergyWindowScan with same parameters, output should be the same.

Repeat for both algorithms with GroupingMethod=`Individual` and MSDFit checked.

Change instrument to `OSIRIS` and InputFiles `osi100318,osi100319,osi100320,osi100321,osi100322` replacing SpectraRange with `963, 1004`
Use RunNumbers `100318-100322` for IndirectEnergyWindowScan

Fixes #14616.

[Release Notes](https://github.com/mantidproject/mantid/blob/14616_IndirectQuickRun_algorithm/docs/source/release/v3.9.0/indirect_inelastic.rst)

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
